packages/ui-navigation/src/AppNav/index.js

```diff
@@ -251,35 +260,30 @@ class AppNav extends Component {
     const renderBeforeItems = callRenderProp(this.props.renderBeforeItems)
     const renderAfterItems = callRenderProp(this.props.renderAfterItems)
     const hasRenderedContent = renderBeforeItems || renderAfterItems
-
     return (
       <View
         {...passthroughProps}
         as="nav"
-        className={classnames({
-          [styles.root]: true,
-          [styles['root--hasRenderedContent']]: hasRenderedContent
-        })}
+        css={[
+          this.props.styles.appNav,
+          hasRenderedContent ? this.props.styles.alignCenter : ''
```

Comment:
Do you think the hasRenderedContent (and renderBeforeItems and renderAfterItems) could be used as a getter, so we can pass it to the makeStyles above?

Reply:
Unfortunately not. we can not call them in componentDidMount , because they do some sort of prop or state changes under the hood and push React into an infinite cycle.

Reply:
Oh, too bad :( Thanks :)